lib/credentials/serviceClientCredentials.ts

```diff
@@ -11,4 +12,12 @@ export interface ServiceClientCredentials {
    * @returns {Promise<WebResource>} The signed request object;
    */
   signRequest(webResource: WebResource): Promise<WebResource>;
+
+  /**
+   * Signs a request with the Authentication header.
+   *
+   * @param request The HTTP request to be signed.
+   * @returns The signed request;
+   */
+  signHttpRequest(httpRequest: HttpRequest): Promise<HttpRequest>;
```

Comment:
TypeScript and Javascript rely on duck typing. As long as the shape remains the same then it should make no difference to consumers. Eventually WebResource will be replaced by HttpRequest. So why have two methods over here? Wont it confuse customers of the node sdk (using ms-rest) today? The runtime is being used a lot by folks inside Microsoft (for developing vscode extensions, etc.) and outside customers as well.

Reply:
See my previous comment.